lib/widgets/content.dart

```diff
@@ -465,43 +466,126 @@ class MessageImageEmoji extends StatelessWidget {
   }
 }
 
+/// Like [Image.network], but includes [authHeader] if [src] is on-realm.
+///
+/// Use this to present image content in the ambient realm: avatars, images in
+/// messages, etc. Must have a [PerAccountStoreWidget] ancestor.
+///
+/// If [src] is an on-realm URL (it has the same origin as the ambient
+/// [Auth.realmUrl]), then an HTTP request to fetch the image will include the
+/// user's [authHeader].
+///
+/// If [src] is off-realm (e.g., a Gravatar URL), no auth header will be sent.
+///
+/// The image will be cached according to the cache behavior of [Image.network],
+/// which may mean the cache is shared between realms.
+class RealmContentNetworkImage extends StatelessWidget {
+  const RealmContentNetworkImage(this.src, {
+    super.key,
```

Comment:
nit:
```suggestion
  const RealmContentNetworkImage(
    this.src, {
    super.key,
```

As is, I missed that first element of the list the first time or two I looked — it kind of blends in.

This alternate version borrows the same formatting used for `Image.network`.